Parcelize: Add an intrinsic to access Parcelable CREATOR fields (KT-19853)

sfs opened the pull request:
The CREATOR field is a static field on a `Parcelable` class which is not visible from Kotlin since the necessary metadata would have to be in a Companion object which may not exist.

This commit adds a `parcelableCreator` function to kotlinx.parcelize, which is optimized to a direct field access whenever possible.

yanex commented:
Looks great!
Am I right that `parcelableCreator()` is going to be mentioned on https://developer.android.com/kotlin/parcelize or somewhere else in the docs?

sfs commented:
Rebased on master and updated test expectations after ae27be16eb20291ef3eff323939a921f9f5181b2 was merged.

> Am I right that `parcelableCreator()` is going to be mentioned on https://developer.android.com/kotlin/parcelize or somewhere else in the docs?

Yes, exactly. :)
I'll reach out after the PR is merged to get something ready for when we have a release including this function.

udalov commented:
Thanks!

IvoB1987 commented:
I was wondering. What's the status of this functionality? No mention of it in the parcelize documentation and it also doesn't seem to work yet, so I guess this functionality is not release yet. Am I correct? Any plans for it?
